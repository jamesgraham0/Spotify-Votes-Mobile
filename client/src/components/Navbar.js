import React from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { MaterialIcons } from '@expo/vector-icons';
import { Ionicons } from '@expo/vector-icons'; 
import Playing from './Playing';
import Search from './Search';
import Queue from './Queue';

const Tab = createBottomTabNavigator();

const Navbar = ({ user, room }) => {
  if (room) {
    return (
      <Tab.Navigator
      initialRouteName="Playing"
      screenOptions={{
        tabBarActiveTintColor: '#1DB954',
        tabBarStyle: {
          backgroundColor: 'black',
          borderTopColor: 'black',
          borderStartColor: '#B026FF',
          borderStartWidth: 1,
          borderEndColor: '#B026FF',
          borderEndWidth: 1,
          borderBottomStartRadius: 80,
          borderBottomEndRadius: 80,
        }
      }}
      >
      <Tab.Screen
        name="Search"
        children={() => <Search room={room}/>}
        options={{
          headerShown: false,
          tabBarLabel: 'Search',
          tabBarIcon: ({ color, size }) => (
            <Ionicons name="search" color={color} size={size} />
            ),
          }}
          />
      <Tab.Screen
        name="Playing"
        children={() => <Playing user={user} room={room} />}
        options={{
          headerShown: false,
          tabBarIcon: ({ color, size }) => (
            <MaterialCommunityIcons name="music-note-plus" color={color} size={size} />
            ),
          }}
          />
      <Tab.Screen
        name="Queue"
        children={() => <Queue queue={room.queue} roomId={room.id} />}
        options={{
          headerShown: false,
          tabBarLabel: 'Queue',
          tabBarIcon: ({ color, size }) => (
            <MaterialIcons name="queue" color={color} size={size} />
            ),
          }}
          />
    </Tab.Navigator>
  );
}
}

export default Navbar;