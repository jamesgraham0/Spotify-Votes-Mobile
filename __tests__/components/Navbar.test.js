import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../../client/src/components/Navbar';

describe('Navbar', () => {
    it('should render the Navbar component', () => {
        const tree = renderer.create(<Navbar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});