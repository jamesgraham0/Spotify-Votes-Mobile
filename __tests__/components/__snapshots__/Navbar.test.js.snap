// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the Navbar component 1`] = `null`;
